### PR TITLE
feat: numeric enum option generation

### PR DESCRIPTION
**Pull request by noahdietz**
Adds support for the `rest-numeric-enums` plugin option that enables generation of the `numericEnums` flag in the REST descriptor config supported in https://github.com/googleapis/gax-php/pull/395.

### CODE REVIEW

**Comment by noahdietz** [comment]
@bshaffer and @dwsupplee this is ready for review now

File: src/CodeGenerator.php [hunk 1]
```diff
@@ -91,6 +93,7 @@ public static function generateFromDescriptor(
      * @param ?string $grpcServiceConfigJson Optional grpc-serv-config json string.
      * @param ?string $gapicYaml Optional gapic configuration yaml string.
      * @param ?string $serviceYaml Optional service configuration yaml string.
+     * @param bool $numericEnums Optional whether to generate the numeric-enums JSON encoding system parameter.
```
Comment (vchudnov-g) [inline src/CodeGenerator.php:96]: nit: consider expanding this a little for clarity: "whether to include in requests the system parameter asking for JSON-encoded responses to encode enum values as numbers"

File: rules_php_gapic/php_gapic.bzl [hunk 2]
```diff
@@ -70,9 +71,13 @@ def php_gapic_srcjar(
     if transport != "grpc+rest" and transport != "rest":
         fail("Error: Only 'grpc+rest' or 'rest' transports are supported")
 
+
     # Set plugin arguments.
     plugin_args = ["metadata"]  # Generate the gapic_metadata.json file.
     plugin_args.append("transport=%s" % transport)
+    
+    if rest_numeric_enums: # Generate REGAPIC param for enum response encoding.
```
Comment (vchudnov-g) [inline rules_php_gapic/php_gapic.bzl:79]: nit: "requesting response enums be JSON-encoded as numbers, not strings"

**Review by vchudnov-g — COMMENTED** [review]
looks good AFAICT, but I'll let PHP generator experts approve.

File: src/CodeGenerator.php (hunk 1 above)
Comment (noahdietz) [reply src/CodeGenerator.php]: Done.

File: rules_php_gapic/php_gapic.bzl (hunk 2 above)
Comment (noahdietz) [reply rules_php_gapic/php_gapic.bzl]: Done.

**Pull request closed without merging**